@xen-orchestra/lite/src/libs/xapi.ts

```diff
@@ -198,4 +212,45 @@ export default class XapiConnection extends EventEmitter {
 
     return _xapi.call(method, ...args)
   }
+
+  async createNetworks(
+    newNetworks: [
+      {
+        bondMode?: 'balance-slb' | 'active-backup' | 'lacp'
+        MTU: number
+        name_description: string
+        name_label: string
+        pifsId?: string[]
+        VLAN: number
+      }
+    ]
+  ): Promise<void[]> {
+    const pifs = this.objectsByType.get('PIF')
+    return Promise.all(
+      newNetworks.map(async ({ bondMode, pifsId, ...newNetwork }) => {
+        let networkRef: string | undefined
+        try {
+          networkRef = (await this.call('network.create', {
+            ...newNetwork,
+            other_config: { automatic: 'false' },
+          })) as string
+
+          if (bondMode !== undefined && pifsId !== undefined) {
+            await Promise.all(
+              pifsId.map(pifId => this.call('Bond.create', networkRef, pifs?.get(pifId)?.$network.PIFs, '', bondMode))
+            )
+            return
+          }
+          if (pifsId !== undefined) {
+            await this.call('pool.create_VLAN_from_PIF', pifs?.get(pifsId[0])?.$ref, networkRef, newNetwork.VLAN)
+          }
```

Comment:
```js
         if (pifIds === undefined){
            // return network Id
          }
          if (bondMode !== undefined) {
            await Promise.all(
              pifsId.map(pifId => this.call('Bond.create', networkRef, pifs?.get(pifId)?.$network.PIFs, '', bondMode))
            )
          } else {
            await this.call('pool.create_VLAN_from_PIF', pifs?.get(pifsId[0])?.$ref, networkRef, newNetwork.VLAN)
          }

          // return networkId
```

`createNetwork` should return the id or ref of the new network.

ping @pdonias

Reply:
Yes.